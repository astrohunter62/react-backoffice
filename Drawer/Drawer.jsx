import React from 'react'
import PropTypes from 'prop-types'

import { withStyles } from 'material-ui/styles'
import MaterialDrawer from 'material-ui/Drawer'
import Divider from 'material-ui/Divider'
import IconButton from 'material-ui/IconButton'
import ChevronLeftIcon from 'material-ui-icons/ChevronLeft'

import Menu from '../Menu'

const drawerWidth = 280

const styles = (theme) => ({
  drawerPaper: {
    position: 'relative',
    height: '100%',
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    height: theme.spacing.unit * 8
  },
})

const Drawer = ({
  data,
  open,
  handleDrawerClose,
  redirectTo,
  classes
}) => {
  return (
    <MaterialDrawer
      type="persistent"
      classes={{
        paper: classes.drawerPaper,
      }}
      anchor="left"
      open={open}
    >
      <div className={classes.drawerInner}>
        <div className={classes.drawerHeader}>
          <IconButton onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
        </div>

        <Divider />

        <Menu data={data} redirectTo={redirectTo} />
      </div>
    </MaterialDrawer>
  )
}

Drawer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  open: PropTypes.bool.isRequired,
  handleDrawerClose: PropTypes.func.isRequired,
  redirectTo: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Drawer)
