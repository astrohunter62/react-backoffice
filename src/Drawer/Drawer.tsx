import React from "react";
import {
  Drawer as MaterialDrawer,
  Divider,
  IconButton,
  withStyles,
} from "@material-ui/core";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import Menu from "../Menu";
import { MenuDataItem } from "../Menu/Menu";
const drawerWidth = 280;
const styles = (theme: any) => ({
  drawerPaper: {
    position: "relative",
    height: "100%",
    width: drawerWidth,
  },
  drawerHeader: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
    height: theme.spacing(8),
  },
});

type DrawerProps = {
  data: MenuDataItem[];
  isOpen?: boolean;
  onClose: (...args: any[]) => any;
  redirectTo: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};

const Drawer: React.SFC<DrawerProps> = ({
  data,
  isOpen,
  onClose,
  redirectTo,
  classes,
}) => (
  <MaterialDrawer
    variant="persistent"
    classes={{
      paper: classes.drawerPaper,
    }}
    anchor="left"
    open={isOpen}
  >
    <div className={classes.drawerInner}>
      <div className={classes.drawerHeader}>
        <IconButton onClick={onClose}>
          <ChevronLeftIcon />
        </IconButton>
      </div>

      <Divider />

      <Menu data={data} redirectTo={redirectTo} />
    </div>
  </MaterialDrawer>
);
Drawer.defaultProps = {
  isOpen: false,
};
export default withStyles(styles as any)(Drawer);
