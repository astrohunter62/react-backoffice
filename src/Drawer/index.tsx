import React from "react";
import {
  Drawer as MaterialDrawer,
  Divider,
  IconButton,
  Theme,
} from "@material-ui/core";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import Menu from "../Menu";
import { MenuDataItem } from "../Menu";
import { makeStyles } from "@material-ui/styles";

export const DRAWER_WIDTH = 240;
const useStyles = makeStyles((theme: Theme) => ({
  drawerPaper: {
    width: DRAWER_WIDTH,
  },

  drawerHeader: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
    height: theme.spacing(8),
  },
}));

type DrawerProps = {
  data: MenuDataItem[];
  isOpen?: boolean;
  onClose: (...args: any[]) => any;
  redirectTo: (...args: any[]) => any;
};

const Drawer: React.SFC<DrawerProps> = ({
  data,
  isOpen = false,
  onClose,
  redirectTo,
}) => {
  const classes = useStyles();

  return (
    <MaterialDrawer
      variant="persistent"
      classes={{
        paper: classes.drawerPaper,
      }}
      anchor="left"
      open={isOpen}
    >
      <div className={classes.drawerHeader}>
        <IconButton onClick={onClose}>
          <ChevronLeftIcon />
        </IconButton>
      </div>

      <Divider />

      <Menu data={data} redirectTo={redirectTo} />
    </MaterialDrawer>
  );
};

export default Drawer;
