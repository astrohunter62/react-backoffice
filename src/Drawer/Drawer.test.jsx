import React from 'react'
import renderer from 'react-test-renderer'

import menuData from '../../__tests__/data/menu'

import Drawer from './'

it('renders correctly', () => {
  const tree = renderer
    .create(<Drawer
      isOpen
      data={menuData}
      onClose={() => {}}
      redirectTo={() => {}}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
