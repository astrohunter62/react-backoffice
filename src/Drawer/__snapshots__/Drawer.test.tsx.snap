// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Drawer
  classes={
    Object {
      "drawerHeader": "Drawer-drawerHeader-2",
      "drawerPaper": "Drawer-drawerPaper-1",
    }
  }
  data={
    Array [
      Object {
        "title": "Dashboard",
        "type": "link",
        "url": "/",
      },
      Object {
        "isDisabled": true,
        "title": "New portfolio",
        "type": "link",
        "url": "/portfolio",
      },
      Object {
        "type": "divider",
      },
      Object {
        "icon": <React.Fragment>
          <UNDEFINED />
        </React.Fragment>,
        "title": "Report a bug",
        "type": "link",
        "url": "/bug",
      },
    ]
  }
  isOpen={true}
  onClose={[Function]}
  redirectTo={[Function]}
/>
`;
